### ISSUE TITLE
Data sometimes inconsistent after it has just changed

### ISSUE THREAD

**Issue by davesumter** [INFURA/infura#61]
I think this is just the normal behavior of multiple nodes synchronizing to a new state, but wanted to check. I sometimes get the wrong data back from infura very shortly after it has changed. Here's what happens:

1) State changes in the blockchain
2) I query Infura and see the new state
3) I query again a few seconds later and see the old state
4) I query again a few seconds after that and see the new correct state again.
5) Correct state from thereon

Is it possible I'm just hitting a different node on the second query which hasn't synchronized yet..? If this is the case I'll adapt my methodology.. But if all my calls are going to the same node then there may be a problem. How many node site behind this endpoint?

This is on mainnet.infura.io

Thanks
Dave

**Comment by tueric** [comment]
Yes, it could be that the new information hasn't been propagated yet to all the nodes in our cluster. Your calls are most likely not hitting the same node.

**Comment by davesumter** [comment]
Thanks Eric, makes sense..

Are you able to share any high level info here..? Not a problem if you can't..

i.e. Roughly how long do you think it takes all nodes to sync to the latest block? Right now I'm seeing it at just few seconds (maybe 5 to 10 in worst case), but I've only made a handful of tests so can't be sure...

I'll use a back-off timer to allow for syncing..

Thanks

**Comment by hhppkjhb** [comment]
@davesumter This is a feature "bug" of Infura given the way that it is architected. They have built their infrastructure using a load balancer in front of multiple ETH nodes, which means that your requests will not always hit the same node. I do not know how long it will take for all their nodes to sync, however, our team solved this issue by switching to a different node provider. We currently use [Alchemy](https://alchemyapi.io/) and haven't had any inconsistency issues since the switch
